trade-finance-manager-ui/server/controllers/case/amendments/amendIsUsingFacilityEndDate.controller.js

```diff
@@ -0,0 +1,78 @@
+const { isTfmFacilityEndDateFeatureFlagEnabled } = require('@ukef/dtfs2-common');
+const api = require('../../../api');
+const { AMENDMENT_STATUS } = require('../../../constants/amendments');
+const { isUsingFacilityEndDateValidation } = require('./validation/amendmentIsUsingFacilityEndDate.validate');
+
+const getAmendIsUsingFacilityEndDate = async (req, res) => {
+  const { facilityId, amendmentId } = req.params;
+  const { userToken } = req.session;
+  const { data: amendment, status } = await api.getAmendmentById(facilityId, amendmentId, userToken);
+
+  if (status !== 200) {
+    return res.redirect('/not-found');
+  }
+
+  const isEditable = amendment.status === AMENDMENT_STATUS.IN_PROGRESS && amendment.changeCoverEndDate && isTfmFacilityEndDateFeatureFlagEnabled();
+  const { dealId, isUsingFacilityEndDate } = amendment;
+
+  return res.render('case/amendments/amendment-is-using-facility-end-date.njk', {
+    dealId,
+    facilityId,
+    isEditable,
+    isUsingFacilityEndDate,
+    user: req.session.user,
+  });
+};
+
+const getBooleanFromIsUsingFacilityEndDate = (isUsingFacilityEndDate) => {
```

Comment:
It's not within the tfm side - similar logic is used in an inline way in other amendment controllers but I think it is a bit dodgy, e.g: 
`const approval = requireUkefApproval === 'Yes'`, so it would mark it as false on e.g. the summary page even if the user has somehow skipped the page / it's undefined.